### PR TITLE
master: enable logical datapath groups for OVN >= 20.12

### PR DESCRIPTION
**Pull request by dcbw**
Logical datapath groups reduce the size of the southbound
database in large clusters. The option is ignored if unsupported
by ovn-northd. ovn-controllers should be upgraded to a version
that supports this option before it is turned on.

@trozet @girishmg @abhat @vpickard

### CODE REVIEW

File: go-controller/pkg/ovn/master.go [hunk 1]
```diff
@@ -253,6 +253,17 @@ func (oc *Controller) StartClusterMaster(masterNodeName string) error {
 		_ = oc.nodeLocalNatIPv6Allocator.Allocate(net.ParseIP(types.V6NodeLocalDistributedGWPortIP))
 	}
 
+	// Enable logical datapath groups on OVN 20.12 and later; option is
+	// ignored if OVN doesn't understand it. Logical datapath groups reduce
+	// the side of the southbound database in large clusters. ovn-controllers
+	// should be upgraded to a version that supports them before the option
+	// is turned on by the master.
+	_, stderr, err := util.RunOVNNbctl("set", "NB_Global", ".", "options:use_logical_dp_groups=true")
```
Comment (abhat) [inline go-controller/pkg/ovn/master.go:261]: Does it make sense to use `NBGlobalSetOptions` API of go-ovn bindings here?

**Review by abhat — CHANGES_REQUESTED** [review]

**Comment by trozet** [comment]
holding why we wait for a version capable of exercising this

File: go-controller/pkg/ovn/master.go [hunk 2]
```diff
@@ -253,6 +253,17 @@ func (oc *Controller) StartClusterMaster(masterNodeName string) error {
 		_ = oc.nodeLocalNatIPv6Allocator.Allocate(net.ParseIP(types.V6NodeLocalDistributedGWPortIP))
 	}
 
+	// Enable logical datapath groups on OVN 20.12 and later; option is
+	// ignored if OVN doesn't understand it. Logical datapath groups reduce
+	// the side of the southbound database in large clusters. ovn-controllers
```
Comment (girishmg) [inline go-controller/pkg/ovn/master.go:258]: NIT: s/side/size/

File: go-controller/pkg/ovn/master.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/ovn/master.go:261]: @abhat sure, why not :)

File: go-controller/pkg/ovn/master.go (hunk 2 above)
Comment (dcbw) [reply go-controller/pkg/ovn/master.go]: fixed

**Comment by abhat** [comment]
> holding while we wait for a version capable of exercising this

It seems like even if there is no version of ovn exercising this, it's an opaque option for OVN that doesn't understand it?

**Comment by trozet** [comment]
control-plane failure is egress ip, already pinged @alexanderConstantinescu on it

dual stack conversion I'm assuming is the known flake. @aojea can you look into this and make a fix please?

**Review by trozet — APPROVED** [review]

**Pull request merged**